### DISCUSSION TITLE
Favicon Forbidden After Update to v2.3.16

### DISCUSSION THREAD

**Discussion by chadgriff** [Budibase/budibase#9801] [Help] [answered]
I have an older app that I recently imported into my production environment. As a result it was updated to v2.3.16.  The home page is taking forever to load (perhaps a min or more) but does finally appear.  My prod environment is self-hosted on a Digital Ocean droplet running Ubuntu. I am using Google Chrome for my browser. Looking at Developer Tools it appears there is an issue with the favicon. At some point in my history, the n8n favicon got related to my Budibase instance (ref attached screen shot). All of the other pages in my app are loading slower but no where near as slow as the home screen. At one point I did get a notification that the Budibase Client library could not be loaded and to try republishing the app.  I republished the app to no avail. Any help would be appreciated.

![favicon_issue_1a](https://user-images.githubusercontent.com/15015476/221228128-c6595645-4359-4331-81f5-536be33c3f64.png)
![favicon_issue_2](https://user-images.githubusercontent.com/15015476/221226891-c8a96420-60f6-4691-b28a-2395391e8d6d.png)

**Comment by melohagan** [comment]
Hey @chadgriff 

The favicon is probably unrelated to the slow down you are experiencing. This may be caused by issues with the skeleton loader, which will be resolved in the upcoming release in the next week or two.

Did you upgrade from version **2.1.x** by any chance? If you rolled back to version **2.1.x** does that help?

**Comment by chadgriff** [comment, marked as answer]
@melohagan I believe you are correct. I have played around with the home page a bit more and realized a data provider nested within a data provider was causing extensive delays in loading. Since the nested data provider only returns a single row I was able to limit that DP to just 1 and that improved the load time significantly. I am encouraged to hear about the forthcoming updates to the skeleton loader. FYI...my app was upgraded from 2.0.37 and yes, in my dev environment, when I rolled it back the performance issues went away.
